codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.go.codegen;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.TriConsumer;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.utils.SmithyBuilder;
+
+/**
+ * Registers additional AWS specific client configuration fields
+ */
+public class AddAwsConfigFields implements GoIntegration {
+    private static final String REGION_CONFIG_NAME = "Region";
+    private static final String CREDENTIALS_CONFIG_NAME = "Credentials";
+    private static final String ENDPOINT_RESOLVER_CONFIG_NAME = "EndpointResolver";
+    private static final String HTTP_CLIENT_CONFIG_NAME = "HTTPClient";
+    private static final String LOGGER_CONFIG_NAME = "Logger";
+    private static final String LOG_LEVEL_CONFIG_NAME = "LogLevel";
+    private static final String RETRYER_CONFIG_NAME = "Retryer";
+
+    private static final List<ConfigField> UNIVERSAL_FIELDS = new ArrayList<>();
+
+    static {
+        UNIVERSAL_FIELDS.add(ConfigField.builder(REGION_CONFIG_NAME, getUniversalSymbol("string"))
```

Comment:
`SetUtils.of` into a `TreeSet` would work a bit better here

Reply:
Good call, changed!